package/MDAnalysis/analysis/pca.py

```diff
@@ -0,0 +1,224 @@
+    >>> pca_space = PSF_pca.transform(n_components=n_pcs)
+
+From here, inspection of the pca_space and conclusions to be drawn from the
+data are left to the user.
+"""
+from six.moves import range
+import logging
+import warnings
+
+import numpy as np
+from MDAnalysis import Universe
+
+from .base import AnalysisBase
+
+logger = logging.getLogger(__name__)
+
+
+class PCA(AnalysisBase):
+    """Principal component analysis on an MD trajectory
+
+    Attributes
+    ----------
+    p_components: array, (n_components, n_atoms)
+        The principal components of the feature space,
+        representing the directions of maximum variance in the data.
+    variance : array (n_components, )
+        The raw variance explained by each eigenvector of the covariance
+        matrix.
+    explained_variance : array, (n_components, )
+        Percentage of variance explained by each of the selected components.
+        If a subset of components is not chosen then all components are stored
+        and the sum of explained variances is equal to 1.0.
+
+    Methods
+    -------
+    fit(traj=None, select='All', start=None, stop=None, step=None)
+        Find the principal components of selected atoms in a subset of the
+        trajectory.
+    transform(traj=None, n_components=None)
+        Take the original trajectory and project it onto the principal
+        components.
+    inverse_tranform(pca_space)
+        Take a pca_space and map it back onto the trajectory used to create it.
+    """
+
+    def __init__(self, atomgroup, demean=True, reference=None,
+                 n_components=None, **kwargs):
+        """
+        Parameters
+        ----------
+        atomgroup: MDAnalysis atomgroup
+            AtomGroup to be used for PCA.
+        n_components : int, optional
+            The number of principal components to be saved, default saves
+            all principal components, Default: -1
+        start : int, optional
+            First frame of trajectory to use for generation
+            of covariance matrix, Default: None
+        stop : int, optional
+            Last frame of trajectory to use for generation
+            of covariance matrix, Default: None
+        step : int, optional
+            Step between frames of trajectory to use for generation
+            of covariance matrix, Default: None
+        """
+        super(PCA, self).__init__(atomgroup.universe.trajectory,
+                                  **kwargs)
+        self._u = atomgroup.universe
+        self.start = kwargs.get('start')
+        self.stop = kwargs.get('stop')
+        self.step = kwargs.get('step')
+
+        if demean:
+            if frame is None:
+                self.reference = self._u
+            else:
+                # change u to traj number for fitting
+                self._u.trajectory[frame]
+                self.reference = self._u
+            aligned = align.AlignTraj(self.reference, self._u).run()
```

Comment:
This is not good. It means the user has to wait a long time when he creates a PCA class and potentially runs out of disk space (think of a 100GB trajectory to be analyzed). The alignment should be done on the fly in `_single_frame`. You can use one of the alignment functions to align a single structure.